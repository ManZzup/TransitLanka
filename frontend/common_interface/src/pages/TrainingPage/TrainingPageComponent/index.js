import React from 'react';
import PlacesAutocompleteContainer from '../../../containers/PlacesAutocompleteContainer';
import ResultsContainer from '../../../containers/ResultsContainer';
import TrainingButtonContainer from '../../../containers/TrainingButtonContainer';

var style = {
  wrapper : {
  },
  column_1 : {
    height:'100vh',
    boxShadow: '2px 1px 3px 1px #ccc',
  },
  heading : {
    fontFamily: ['Open Sans Condensed', 'sans-serif'],
    marginBottom:'-10px'
  },
  sub_heading : {
    fontSize: '20px',
    backgroundColor:'#ffae00',
    padding:'5px',
    color:'#fff'
  },
  heading2 : {
    fontFamily: ['Open Sans Condensed', 'sans-serif'],
    fontSize: '28px',
    color: '#1C6BA0',
    fontWeight: 'bold'
  },
  heading3 : {
    fontSize: '20px'
  },
};

const TrainingPageComponent = (props) => (
    <div className="">
      <div className="large-4 small-12 columns"  style={style.column_1}>
          <div style={style.wrapper}>
            <h1 style={style.heading}> <font color="#1C6BA0">transit</font><font color="#ffae00">Lanka</font></h1>
            <span style={style.sub_heading}>trainMe</span>
            <br /><br />

            {props.hasTrainingSet &&
                <div>
                  Which option out of the shown routes you will take to travel <br /><br />

                  <div className="row">
                    <div className="input-group">
                      <span className="input-group-label">From</span>
                      <input className="input-group-field awesomplete" type="text"
                             readOnly
                             value={props.startLocation} />
                    </div>
                  </div>
                  <div className="row">
                    <div className="input-group">
                      <span className="input-group-label">To</span>
                      <input className="input-group-field awesomplete" type="text"
                             readOnly
                             value={props.endLocation} />
                    </div>
                  </div>

                </div>
            }

            {!props.hasTrainingSet &&

              <p>
                Hi, welcome to training session of transitLanka! <br />
                To begin please press the button below
              </p>
            }

            <TrainingButtonContainer />

            {props.hasTrainingSet &&
              <div>
                <hr />
                <h2 style={style.heading2}>FAQ</h2>
                <div>
                  <h3 style={style.heading3}><font color="#ffae00">Q.</font> How to select an option?</h3>
                  <p><font color="#1C6BA0">A.</font>  Click the âœ” Button to accept the option</p>
                </div>
                <div>
                  <h3 style={style.heading3}><font color="#ffae00">Q.</font> What if I dont know the answer?</h3>
                  <p><font color="#1C6BA0">A.</font>  Press the "New Training Set" button to get a new set</p>
                </div>
              </div>
            }

          </div>


      </div>
      <div className="large-7 small-12 columns end">
          <ResultsContainer />
      </div>
    </div>
);

export default TrainingPageComponent;
