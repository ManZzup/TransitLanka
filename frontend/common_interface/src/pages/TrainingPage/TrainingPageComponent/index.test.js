import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux'
import { createStore,applyMiddleware } from 'redux'
import thunkMiddleware from 'redux-thunk'
import TrainingPageComponent from './';
import rootReducer from '../../../reducers';

let store = createStore(rootReducer,applyMiddleware(
    thunkMiddleware
));

it('renders Training Page without Training Set correctly', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <TrainingPageComponent startText="Pettah" endText="Moratuwa" hasTrainingSet={false} />
    </Provider>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders Training Page with Training Set correctly', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <TrainingPageComponent startText="Pettah" endText="Moratuwa" hasTrainingSet={true} />
    </Provider>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
