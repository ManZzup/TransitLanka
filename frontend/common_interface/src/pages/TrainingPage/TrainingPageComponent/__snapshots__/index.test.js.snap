exports[`test renders Training Page with Training Set correctly 1`] = `
<div
  className="">
  <div
    className="large-4 small-12 columns"
    style={
      Object {
        "boxShadow": "2px 1px 3px 1px #ccc",
        "height": "100vh",
      }
    }>
    <div
      style={Object {}}>
      <h1
        style={
          Object {
            "fontFamily": Array [
              "Open Sans Condensed",
              "sans-serif",
            ],
            "marginBottom": "-10px",
          }
        }>
         
        <font
          color="#1C6BA0">
          transit
        </font>
        <font
          color="#ffae00">
          Lanka
        </font>
      </h1>
      <span
        style={
          Object {
            "backgroundColor": "#ffae00",
            "color": "#fff",
            "fontSize": "20px",
            "padding": "5px",
          }
        }>
        trainMe
      </span>
      <br />
      <br />
      <div>
        Which option out of the shown routes you will take to travel 
        <br />
        <br />
        <div
          className="row">
          <div
            className="input-group">
            <span
              className="input-group-label">
              From
            </span>
            <input
              className="input-group-field awesomplete"
              readOnly={true}
              type="text"
              value={undefined} />
          </div>
        </div>
        <div
          className="row">
          <div
            className="input-group">
            <span
              className="input-group-label">
              To
            </span>
            <input
              className="input-group-field awesomplete"
              readOnly={true}
              type="text"
              value={undefined} />
          </div>
        </div>
      </div>
      <div
        className="row">
        <input
          className="button warning"
          onClick={[Function]}
          type="button"
          value="New Training Set" />
      </div>
      <div>
        <hr />
        <h2
          style={
            Object {
              "color": "#1C6BA0",
              "fontFamily": Array [
                "Open Sans Condensed",
                "sans-serif",
              ],
              "fontSize": "28px",
              "fontWeight": "bold",
            }
          }>
          FAQ
        </h2>
        <div>
          <h3
            style={
              Object {
                "fontSize": "20px",
              }
            }>
            <font
              color="#ffae00">
              Q.
            </font>
             How to select an option?
          </h3>
          <p>
            <font
              color="#1C6BA0">
              A.
            </font>
              Click the âœ” Button to accept the option
          </p>
        </div>
        <div>
          <h3
            style={
              Object {
                "fontSize": "20px",
              }
            }>
            <font
              color="#ffae00">
              Q.
            </font>
             What if I dont know the answer?
          </h3>
          <p>
            <font
              color="#1C6BA0">
              A.
            </font>
              Press the \"New Training Set\" button to get a new set
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    className="large-7 small-12 columns end">
    <div
      className="small-12 columns"
      id="results"
      style={
        Object {
          "paddingTop": "10px",
        }
      }>
      <span />
    </div>
  </div>
</div>
`;

exports[`test renders Training Page without Training Set correctly 1`] = `
<div
  className="">
  <div
    className="large-4 small-12 columns"
    style={
      Object {
        "boxShadow": "2px 1px 3px 1px #ccc",
        "height": "100vh",
      }
    }>
    <div
      style={Object {}}>
      <h1
        style={
          Object {
            "fontFamily": Array [
              "Open Sans Condensed",
              "sans-serif",
            ],
            "marginBottom": "-10px",
          }
        }>
         
        <font
          color="#1C6BA0">
          transit
        </font>
        <font
          color="#ffae00">
          Lanka
        </font>
      </h1>
      <span
        style={
          Object {
            "backgroundColor": "#ffae00",
            "color": "#fff",
            "fontSize": "20px",
            "padding": "5px",
          }
        }>
        trainMe
      </span>
      <br />
      <br />
      <p>
        Hi, welcome to training session of transitLanka! 
        <br />
        To begin please press the button below
      </p>
      <div
        className="row">
        <input
          className="button warning"
          onClick={[Function]}
          type="button"
          value="New Training Set" />
      </div>
    </div>
  </div>
  <div
    className="large-7 small-12 columns end">
    <div
      className="small-12 columns"
      id="results"
      style={
        Object {
          "paddingTop": "10px",
        }
      }>
      <span />
    </div>
  </div>
</div>
`;
